HeliPort/Appearance/StatusMenu.swift

```diff
@@ -30,9 +30,12 @@ final class StatusMenu: NSMenu, NSMenuDelegate {
     private var networkListUpdateTimer: Timer?
     private var statusUpdateTimer: Timer?
 
+    private var isDisconnecting: Bool = false
```

Comment:
I'm guessing this flag is to prevent race conditions? Justification will be appreciated.

Reply:
This flag serves so that when the user manually disconnects from the network, the condition on line +49 will not attempt to disconnect again and will not call NetworkManager.scanSavedNetworks(), thus allowing the user to remain disconnected.